neo/Consensus/ConsensusService.cs

```diff
@@ -136,13 +148,31 @@ private void Log(string message, LogLevel level = LogLevel.Info)
 
         private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
         {
+            if (context.State.HasFlag(ConsensusState.CommitSent))
+                return;
             if (message.NewViewNumber <= context.ExpectedView[payload.ValidatorIndex])
                 return;
             Log($"{nameof(OnChangeViewReceived)}: height={payload.BlockIndex} view={message.ViewNumber} index={payload.ValidatorIndex} nv={message.NewViewNumber}");
             context.ExpectedView[payload.ValidatorIndex] = message.NewViewNumber;
             CheckExpectedView(message.NewViewNumber);
         }
 
+        private void OnCommitReceived(ConsensusPayload payload, Commit commit)
+        {
+            if (context.Commits[payload.ValidatorIndex] != null) return;
+            Log($"{nameof(OnCommitReceived)}: height={payload.BlockIndex} view={commit.ViewNumber} index={payload.ValidatorIndex}");
+            byte[] hashData = context.MakeHeader()?.GetHashData();
+            if (hashData == null)
+            {
```

Comment:
When could be null, and why skip the check of the signature?

Reply:
You may receive a `Commit` message before receiving the `PrepareRequest` message.

Reply:
But we need to receive both?

Reply:
Yes, but we should check in all cases, otherwise a node can try to relay a block with wrong signature in the last phase, it used to cause an error when I tested but I do not remember exactly, ahauahaha

We can double check this a later.

Reply:
The skiped checks are re-checked at here:

https://github.com/neo-project/neo/blob/939b04fe3bfd563a52b3042bef0ebfd98d77a2c8/neo/Consensus/ConsensusService.cs#L249-L253

Reply:
Gud veri naice, Eriks. Thanks, I forgot about this jump in this design.

The other PR was different strategy.
Anyway, it is good we are all sure about these points.